### PR TITLE
Added 'how to run them' in README.md

### LINKED ISSUE #243: Add instructions on how to run code in README
It would be great if we had instructions on how to run any code present in this repo . We can have this in README.md . 

Example, for Go

```
go run file.go
```

For Python

```
python file.py
```

### PR DESCRIPTION
**Pull request by S2606**
Fixes #243 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous timespan of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi @S2606, thanks for your PR. We're really glad that you showed interest in contributing to this repo.

Make sure that you are assigned to the issue before working on it.

**Comment by S2606** [comment]
@singhpratyush I shall ensure it

File: README.md [hunk 1]
```diff
@@ -50,6 +49,12 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Linked List](https://en.wikipedia.org/wiki/Linked_list) | | | [\[X\]](linked_list/LinkedList.java) | | | |
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)) | | | [\[X\]](stack/Stack.java) | [\[X\]](stack/stack.py) | [\[X\]](stack/stack.go) | [\[X\]](stack/stack.js) |
 
+## How to run them
+| C | CPP | Java | Python | Golang | JavaScript |
+|:----------------:|:----------------:|:----------------:|:-----------------:|:-----------------:|:-----------------:|
+|gcc -o [file.c]| g++ [file.cpp] -o [file]| javac [file.java]| python [file.py]| go run [file.go]|
```
Comment (singhpratyush) [inline README.md]: It is `node [file.js]` for JavaScript.

Enclose the commands in back quote.

File: README.md [hunk 2]
```diff
@@ -38,7 +37,7 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Quicksort](https://en.wikipedia.org/wiki/Quicksort) | [\[X\]](quicksort/quicksort.c) | | [\[X\]](quicksort/QuickSort.java) | [\[X\]](quicksort/quick_sort.py) | | |
 | [Rod Cutting Problem](http://www.geeksforgeeks.org/dynamic-programming-set-13-cutting-a-rod/) | [\[X\]](rod_cutting_problem/rod_cutting.c) | | [\[X\]](rod_cutting_problem/RodCutting.java) | [\[X\]](rod_cutting_problem/rod_cutting.py) | | |
 | [Shell Sort](https://en.wikipedia.org/wiki/Shellsort) | | [\[X\]](shell_sort/ShellSort.cpp) | | [\[X\]](/shell_sort/shell_sort.py) | [\[X\]](shell_sort/shell_sort.go) | |
-| [Sieve of Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes) | [\[X\]](sieve_of_eratosthenes/sieveOfEratosthenes.c) | | [\[X\]](sieve_of_eratosthenes/SieveOfEratosthenes.java) | [\[X\]](sieve_of_eratosthenes/sieve_of_eratosthenes.py) | [\[X\]](sieve_of_eratosthenes/sieve_of_eratosthenes.go) | |
+| [Sieve of Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes) | [\[X\]](sieve_of_eratosthenes/sieveOfEratosthenes.c) | | | [\[X\]](sieve_of_eratosthenes/sieve_of_eratosthenes.py) | [\[X\]](sieve_of_eratosthenes/sieve_of_eratosthenes.go) | |
```
Comment (singhpratyush) [inline README.md]: Why this change? Please revert this.

File: README.md (hunk 1 above)
Comment (S2606) [inline README.md]: ok will do it

File: README.md (hunk 2 above)
Comment (S2606) [inline README.md]: sorry for the mistake.Will revert it

**Comment by S2606** [comment]
@singhpratyush the changes have been made

File: README.md [hunk 3]
```diff
@@ -12,7 +12,6 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 
 ## Implemented Algorithms
 
-
```
Comment (singhpratyush) [inline README.md]: Please do not change this section at all.

File: README.md [hunk 4]
```diff
@@ -50,6 +49,12 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Linked List](https://en.wikipedia.org/wiki/Linked_list) | | | [\[X\]](linked_list/LinkedList.java) | | | |
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)) | | | [\[X\]](stack/Stack.java) | [\[X\]](stack/stack.py) | [\[X\]](stack/stack.go) | [\[X\]](stack/stack.js) |
 
+## How to run them
+| C | CPP | Java | Python | Golang | JavaScript |
+|:----------------:|:----------------:|:----------------:|:-----------------:|:-----------------:|:-----------------:|
+|`gcc -o [filename.c]`| `g++ [filename.cpp] -o [filename]`| `javac [filename.java]`| `python [filename.py]`| `go run [filename.go]`| `node [filename.js]`
```
Comment (singhpratyush) [inline README.md]: Add | in the end.

File: README.md [hunk 5]
```diff
@@ -38,7 +37,7 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
 | [Rod Cutting Problem](http://www.geeksforgeeks.org/dynamic-programming-set-13-cutting-a-rod/) | [\[X\]](rod_cutting_problem/rod_cutting.c) | | [\[X\]](rod_cutting_problem/RodCutting.java) | [\[X\]](rod_cutting_problem/rod_cutting.py) | | |
 | [Shell Sort](https://en.wikipedia.org/wiki/Shellsort) | | [\[X\]](shell_sort/ShellSort.cpp) | | [\[X\]](/shell_sort/shell_sort.py) | [\[X\]](shell_sort/shell_sort.go) | |
-| [Sieve of Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes) | [\[X\]](sieve_of_eratosthenes/sieveOfEratosthenes.c) | | [\[X\]](sieve_of_eratosthenes/SieveOfEratosthenes.java) | [\[X\]](sieve_of_eratosthenes/sieve_of_eratosthenes.py) | [\[X\]](sieve_of_eratosthenes/sieve_of_eratosthenes.go) | |
+| [Sieve of Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes) | [\[X\]](sieve_of_eratosthenes/sieveOfEratosthenes.c) | |[\[X\]](sieve_of_eratosthenes/SieveOfEratosthenes.java) | [\[X\]](sieve_of_eratosthenes/sieve_of_eratosthenes.py) | [\[X\]](sieve_of_eratosthenes/sieve_of_eratosthenes.go) | |
```
Comment (singhpratyush) [inline README.md]: This section is still showing a diff. Please resolve.

**Comment by singhpratyush** [comment]
@aviaryan: Is using tables for this purpose feasible? Would it still be effective if number of languages increase?

File: README.md (hunk 3 above)
Comment (S2606) [inline README.md]: Ok Got it

**Comment by aviaryan** [comment]
@singhpratyush Yes that should work. As far as I think, we only need 2 columns - "Language Name" and steps, so it should be fine.

**Comment by S2606** [comment]
I will make the necessary changes @aviaryan

**Comment by aviaryan** [comment]
@S2606 Sorry I meant that you should make language name and steps to run as columns. So there will be only 2 columns. This will help us avoid overflow when more languages are added.

**Comment by S2606** [comment]
No problem. I will fix it @aviaryan

File: README.md [hunk 6]
```diff
@@ -51,9 +51,14 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)) | | | [\[X\]](stack/Stack.java) | [\[X\]](stack/stack.py) | [\[X\]](stack/stack.go) | [\[X\]](stack/stack.js) |
 
 ## How to run them
-| `gcc -o [filename.c]` | `g++ [filename.cpp] -o [filename]` | `javac [filename.java]` | `python [filename.py]` | `go run [filename.go]` | `node [filename.js]` |
-|:----------------:|:----------------:|:----------------:|:-----------------:|:-----------------:|:-----------------:|
-| C | CPP | Java | Python | Golang | JavaScript |
+| Language | Steps |
+|:--------------|:----------------:|
```
Comment (singhpratyush) [inline README.md]: Do not centre align the second column.

File: README.md [hunk 7]
```diff
@@ -51,6 +51,18 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)) | | | [\[X\]](stack/Stack.java) | [\[X\]](stack/stack.py) | [\[X\]](stack/stack.go) | [\[X\]](stack/stack.js) |
 
 
+## How to run them
+| Language | Steps |
+|:--------------|:----------------:|
+| C | `gcc -o [filename.c]` |
```
Comment (aviaryan) [inline README.md]: Add full steps .. like in C and CPP, add the step to run the output program.
Example

```sh
gcc [filename.c]
./a.out  # unix
./a.exe  # windows
```

Comment (S2606) [inline README.md]: Ok, will update it

File: README.md (hunk 6 above)
Comment (S2606) [inline README.md]: Ok, will update it

File: README.md [hunk 8]
```diff
@@ -51,6 +51,18 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)) | | | [\[X\]](stack/Stack.java) | [\[X\]](stack/stack.py) | [\[X\]](stack/stack.go) | [\[X\]](stack/stack.js) |
 
 
+## How to run them
+| Language | Steps |
+|:--------------|:----------------|
+| C |`gcc [filename.c]`<br>`./filename.out  # unix`<br>`./filename.exe  # windows`|
+| CPP |`g++ [filename.cpp]`<br>`./filename.out # unix`<br>`filename.exe # windows`|
+| Java |`javac [filename.java]`<br>`java filename # windows # unix`|
+| Python |`# for python version < 3`<br>`python [filename.py]`<br>`# for python version > 3`<br>`python3 [filename.py]`<br>`# windows # unix`|
```
Comment (aviaryan) [inline README.md]: It would be better to have multi line code boxes here (triple ticks). Single line `code boxes` look ugly in this context. 
But that might not be possible in markdown table. So use HTML tables. 
Use
```
<table>
<tr>..<td>.. 
</table>
```

Comment (S2606) [inline README.md]: ok,will do that

**Comment by aviaryan** [comment]
Code blocks are still single. I meant something like this. 

| Language | Steps |
|:--------------|:----------------|
| C | <pre>gcc [filename.c] <br>./a.out</pre>|

```
| Language | Steps |
|:--------------|:----------------|
| C | <pre>gcc [filename.c] <br>./a.out</pre>|
```

**Comment by S2606** [comment]
@aviaryan its removed

File: README.md [hunk 9]
```diff
@@ -51,6 +51,19 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)) | | | [\[X\]](stack/Stack.java) | [\[X\]](stack/stack.py) | [\[X\]](stack/stack.go) | [\[X\]](stack/stack.js) |
 
 
+## How to run them
+
+| Language | Steps |
+|:--------------|:----------------|
+| C |<pre>`gcc [filename.c]`<br>`./filename.out  # unix`<br>`./filename.exe  # windows`</pre>|
```
Comment (singhpratyush) [inline README.md]: The binary files generated are `a.exe` and `a.out`.

File: README.md [hunk 10]
```diff
@@ -51,6 +51,19 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
+| CPP |<pre>`g++ [filename.cpp]`<br>`./filename.out # unix`<br>`filename.exe # windows`</pre>|
+| Java |<pre>`javac [filename.java]`<br>`java filename`</pre>|
+| Python |<pre>`# for python version upto 3`<br>`python [filename.py]`<br>`# for python version above 3`<br>`python3 [filename.py]`</pre>|
```
Comment (singhpratyush) [inline README.md]: Just keep it `python [filename.py]`.

File: README.md [hunk 11]
```diff
@@ -51,6 +51,19 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
+| Golang |<pre>`go run [filename.go]`</pre>|
+| JavaScript |<pre>`node [filename.js]`</pre>|
```
Comment (singhpratyush) [inline README.md]: Is `<pre>` required? Will \`...\` not do the job?

File: README.md [hunk 12]
```diff
@@ -51,6 +51,19 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
+
+
+
```
Comment (singhpratyush) [inline README.md]: Keep just two blank lines.

File: README.md (hunk 11 above)
Comment (aashutoshrathi) [inline README.md]: no `<pre>` is good but  but " ` "should be removed as they are not needed.

File: README.md [hunk 13]
```diff
@@ -51,6 +51,18 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)) | | | [\[X\]](stack/Stack.java) | [\[X\]](stack/stack.py) | [\[X\]](stack/stack.go) | [\[X\]](stack/stack.js) |
 
 
+## How to run them
+
+| Language | Steps |
+|:--------------|:----------------|
+| C |<pre>gcc [filename.c]<br>filename.out  # unix<br>filename.exe  # windows</pre>|
+| CPP |<pre>g++ [filename.cpp]<br>./filename.out # unix<br>filename.exe # windows</pre>|
```
Comment (aashutoshrathi) [inline README.md]: change binaries to `a.exe` and `a.bin` as suggested by @singhpratyush

Comment (S2606) [inline README.md]: a.bin for unix right??

Comment (aashutoshrathi) [inline README.md]: Why did you removed `./` ? Add them back.

Comment (aashutoshrathi) [inline README.md]: I'm not sure about UNIX, but in Windows binary formed is `a.exe`

Comment (S2606) [inline README.md]: changes made

Comment (singhpratyush) [inline README.md]: `./a.out` for *nix.

**Comment by S2606** [comment]
@aashutoshrathi @aviaryan @singhpratyush Made all the necessary modification

File: README.md [hunk 14]
```diff
@@ -51,6 +51,18 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)) | | | [\[X\]](stack/Stack.java) | [\[X\]](stack/stack.py) | [\[X\]](stack/stack.go) | [\[X\]](stack/stack.js) |
 
 
+## How to run them
+
+| Language | Steps |
+|:--------------|:----------------|
+| C |<pre>gcc [filename.c]<br>./filename.out  # unix<br>./filename.exe  # windows</pre>|
```
Comment (singhpratyush) [inline README.md]: This still is incorrect. It is `./a.out` for unix and `a.exe` for windows.

File: README.md [hunk 15]
```diff
@@ -51,6 +51,18 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
+| CPP |<pre>g++ [filename.cpp]<br>./filename.out # unix<br>filename.exe # windows</pre>|
+| Java |<pre>javac [filename.java]<br>java filename</pre>|
```
Comment (singhpratyush) [inline README.md]: It should be `java [filename]`.

Comment (S2606) [inline README.md]: ok changed that

File: README.md [hunk 16]
```diff
@@ -51,6 +51,18 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)) | | | [\[X\]](stack/Stack.java) | [\[X\]](stack/stack.py) | [\[X\]](stack/stack.go) | [\[X\]](stack/stack.js) |
 
 
+## How to run them
+
+| Language | Steps |
+|:--------------|:----------------|
+| C |<pre>gcc [filename.c]<br>./filename.out  # unix<br>filename.exe  # windows</pre>|
```
Comment (singhpratyush) [inline README.md]: It IS **a**.out and not filename.out/exe.

File: README.md [hunk 17]
```diff
@@ -51,6 +51,18 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)) | | | [\[X\]](stack/Stack.java) | [\[X\]](stack/stack.py) | [\[X\]](stack/stack.go) | [\[X\]](stack/stack.js) |
 
 
+## How to run them
+
+| Language | Steps |
+|:--------------|:----------------|
+| C |<pre>gcc [filename.c]<br>./a.out  # unix<br>a.exe  # windows</pre>|
+| CPP |<pre>g++ [filename.cpp]<br>./filename.out # unix<br>filename.exe # windows</pre>|
```
Comment (singhpratyush) [inline README.md]: The same goes for C++ section too.

File: README.md [hunk 18]
```diff
@@ -51,6 +51,18 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)) | | | [\[X\]](stack/Stack.java) | [\[X\]](stack/stack.py) | [\[X\]](stack/stack.go) | [\[X\]](stack/stack.js) |
 
 
+## How to run them
+
+| Language | Steps |
+|:--------------|:----------------|
+| C |<pre>gcc [filename.c]<br>./a.out  # unix<br>a.exe  # windows</pre>|
+| CPP |<pre>g++ [filename.cpp]<br>./a.out # unix<br>a.exe # windows</pre>|
```
Comment (aashutoshrathi) [inline README.md]: Please Change `a.exe`  to `./a.exe` as they are different and one isn't working
screenshot 21

**Comment by singhpratyush** [comment]
@aashutoshrathi: Is this problem only with `PowerShell`? Have you tried with command prompt?

**Comment by S2606** [comment]
Thank you @singhpratyush @aviaryan @aashutoshrathi for co-operating with me(and also helping) in merging my first ever PR!!

**Comment by aashutoshrathi** [comment]
@singhpratyush : My bad, this one is `PowerShell` specific 😄 . It's working in `Command Prompt`.
But Command Prompt doesn't support `./a.exe`, so it turns out we can't fix one for windows

**Comment by singhpratyush** [comment]
@S2606: Welcome to the open source world. There are a lot of projects all around Github where you can contribute and learn a lot of things.

If you interested in getting a badge from this org, please open an issue in [iiitv/ShareIt](https://github.com/iiitv/ShareIt) and we'll get you on board :smile:.

**Comment by singhpratyush** [comment]
@aashutoshrathi: My plan was to not keep it there at all (hail Linux :wink:).
